upstairs/src/lib.rs

```diff
@@ -4401,14 +4396,48 @@ impl Upstairs {
         let next_id = downstairs.next_id();
         let mut cur_offset: usize = 0;
 
-        let mut dep = downstairs.active.keys().cloned().collect::<Vec<u64>>();
-        dep.sort_unstable();
+        /*
+         * Walk the downstairs work active list, and pull out all the active
+         * jobs (anything we have not submitted back to the guest). Construct
+         * a list of dependencies for this write based on the
+         * following rules:
+         *
+         * - ignore everything that happened before the last flush
+         * - writes have to depend on the last flush completing
+         * - any overlap of impacted blocks requires a dependency
+         */
+        let mut dep: Vec<u64> = Vec::with_capacity(200); // max in-flight jobs
```

Comment:
If there is a max in-flight number somewhere, we should make it const so anything that needs it can find it.